tests/integration/helpers.py

```diff
@@ -11,3 +26,58 @@ async def get_leader_unit(model, application_name: str) -> Unit:
         if unit.application == application_name and await unit.is_leader_from_status():
             return unit
     raise RuntimeError(f"Leader unit for `{application_name}` not found.")
+
+
+async def get_unit_status_messages(
+    ops_test: OpsTest, app_name: str = APP_NAME
+) -> List[tuple[str, str]]:
+    """Get the status messages from all the units of the given application."""
+    return_code, stdout, stderr = await ops_test.juju("status", "--format", "yaml", app_name)
+    if return_code:
+        raise RuntimeError(stderr)
+    output = yaml.safe_load(stdout)
+    unit_statuses = output["applications"][app_name]["units"]
+    return [
+        (unit_name, unit_status["workload-status"]["message"])
+        for (unit_name, unit_status) in unit_statuses.items()
+    ]
+
+
+async def wait_for_vault_status_message(
+    ops_test: OpsTest,
+    count: int,
+    expected_message: str,
+    timeout: int = 100,
+    cadence: int = 2,
+    app_name: str = APP_NAME,
+) -> None:
+    """Wait for the correct vault status messages to appear.
+
+    This function is necessary because ops_test doesn't provide the facilities
+    to discriminate depending on the status message of the units, just the
+    statuses themselves.
```

Comment:
```suggestion
    application statuses.
```

I'm not sure if my recommendation is right btw but the existing phrasing was confusing to me.

Reply:
Yeah I think what you wrote makes more sense. This function was actually just copied from the k8s charm, so I'll make the change there, too.

Reply:
https://github.com/canonical/vault-k8s-operator/pull/405